### PR TITLE
fix: [CI-9625]: CI credit card onboarding with trust level

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

CI credit card onboarding with trust level

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx [hunk 1]
```diff
@@ -397,25 +397,34 @@ export default function BuildInfraSpecifications({ children }: React.PropsWithCh
   const isFreeEdition = isFreePlan(licenseInformation, ModuleName.CI)
   const isSecurityEnterprise =
     isEnterprisePlan(licenseInformation, ModuleName.STO) && licenseInformation['STO']?.status === 'ACTIVE'
-  const [isCloudInfraDisabled, setIsCloudInfraDisabled] = useState<boolean>(true)
+  const [isCloudInfraDisabled, setIsCloudInfraDisabled] = useState<boolean>(false)
```
Comment (d1wilko) [inline src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:400]: not worth holding up this PR over - but this variable would be easier to work out if it was positive

something like isCloudInfraEnabled

**Review by d1wilko — DISMISSED** [review]
one suggestion - but looks good

**Review by vardanbansal-harness — APPROVED** [review]

**Comment by devkimittal** [comment]
retrigger jest

**Comment by devkimittal** [comment]
retrigger jest

**Comment by devkimittal** [comment]
retrigger jest

**Pull request merged**